### ISSUE TITLE
Refactor/Enhance Tests Setup

### ISSUE THREAD

**Issue by YousefAldabbas** [igorbenav/FastAPI-boilerplate#147]
### Discussed in https://github.com/igorbenav/FastAPI-boilerplate/discussions/146

<div type='discussions-op-text'>

<sup>Originally posted by **YousefAldabbas** July 18, 2024</sup>
Hi all,

I'm planning to open an issue and close it. I want to know the community's/contributors' thoughts before proceeding.

The existing unit tests are relying on each other to pass the unit tests for example `test_update_user` relies  on `test_post_user` success, Which violate unit tests best practices *unit tests should be able to run in isolation* [src for example](https://medium.com/@kaanfurkanc/unit-testing-best-practices-3a8b0ddd88b5).
Also `test_update_user` is calling another function which is relies on the server itself to generate token and then send it with the tests requests , it should be mocked.

issue: usage of _get_token & environment variables to create user for testing 

![image](https://github.com/user-attachments/assets/76affa2b-7505-4a45-93d2-0532f17649e4)

_get_token code 
![image](https://github.com/user-attachments/assets/b53cff4d-2cc8-4f10-a19e-c77cb7c8eb3d)

after implementing applying
```
def test_update_user(db: Session, client: TestClient) -> None:
    user = generators.create_users(db) #generate mocked user
    override_dependancy(auth_dep, user)
    updated_name = f"Updated {user.name}"

    response = client.patch(
        f"/api/v1/user/{user.username}",
        json={"name": updated_name}
    )
    assert response.status_code == 200
    # now we can validate that the data is updated!
    # something like
    # assert response.json()["name"] == updated_name 
```

and override function code (copied from one of my repos)
```
def override_dependency(dependency: Callable[..., Any], mocked_response: Any):
    """
    Overrides a dependency with a mocked response.

    Parameters:
        dependency (Callable[..., Any]): The dependency to override.
        mocked_response (Any): The mocked response to use.

    Returns:
        None

    Example:
        `override_dependency(get_data, mocked_data)`
    """
    app.dependency_overrides[dependency] = lambda: mocked_response
```
after that we can remove these env variable
```
test_name = settings.TEST_NAME
test_username = settings.TEST_USERNAME
test_email = settings.TEST_EMAIL
test_password = settings.TEST_PASSWORD
```
we can also generate username and password for the admin using faker package

theres two new packages i will be adding:
- psycopg2-binary to support db fixture (get_session)
- faker

in the end we will be able to tests all endpoints in **isolation** 
</div>

**Comment by YousefAldabbas** [comment]
@igorbenav  please review the changes, but there's one issue remains in the read me file - project structure section i wasn't able to make changes without destroying the whole project design can you modifies it please ;)
also i couldnt create PR I'm not sure if its related to project settings or issue from my side

**Comment by igorbenav** [comment]
Closed by #148
